sisl/geom/surfaces.py

```diff
@@ -0,0 +1,587 @@
+        out.set_nsc(c=1)
+
+    return out
+
+
+@set_module("sisl.geom")
+def fcc_slab(alat, atoms, miller, layers=None, vacuum=20., *, orthogonal=False, start=None, end=None):
+    r""" Construction of a surface slab from a face-centered cubic (FCC) crystal
+
+    The slab layers are stacked along the :math:`z`-axis. The default stacking is the first
+    layer as an A-layer, defined as the plane containing an atom at :math:`(x,y)=(0,0)`.
+
+    Parameters
+    ----------
+    alat : float
+        lattice constant of the fcc crystal
+    atoms : Atom
+        the atom that the crystal consists of
+    miller : int or str or (3,)
+        Miller indices of the surface facet
+    layers : int or str, optional
+        Number of layers in the slab or explicit layer specification.
+        An empty character `' '` will be denoted as a vacuum slot, see examples.
+        Currently the layers cannot have stacking faults.
+    vacuum : float or array_like, optional
+        distance added to the third lattice vector to separate
+        the slab from its periodic images. If this is None, the slab will be a fully
+        periodic geometry but with the slab layers. Useful for appending geometries together.
+        If an array layers should be a str, it should be no longer than the number of spaces
+        in `layers`. If shorter the last item will be repeated (like `zip_longest`).
+    orthogonal : bool, optional
+        if True returns an orthogonal lattice
+    start : int or string, optional
+        sets the first layer in the slab. Only one of `start` or `end` must be specified.
+        If set together with `layers` being a str, then they *must* be conforming.
+    end : int or string, optional
+        sets the last layer in the slab. Only one of `start` or `end` must be specified.
+        If set together with `layers` being a str, then they *must* be conforming.
+
+    Examples
+    --------
+    111 surface, starting with the A layer
+    >>> fcc_slab(alat, atoms, "111", start=0)
+
+    111 surface, starting with the B layer
+    >>> fcc_slab(alat, atoms, "111", start=1)
+
+    111 surface, ending with the B layer
+    >>> fcc_slab(alat, atoms, "111", end='B')
+
+    111 surface, with explicit layers in a given order
+    >>> fcc_slab(alat, atoms, "111", layers='BCABCA')
+
+    111 surface, with (1 Ang vacuum)BCA(2 Ang vacuum)ABC(3 Ang vacuum)
+    >>> fcc_slab(alat, atoms, "111", layers=' BCA ABC ', vacuum=(1, 2, 3))
+
+    111 surface, with (20 Ang vacuum)BCA
+    >>> fcc_slab(alat, atoms, "111", layers=' BCA', vacuum=20)
+
+    111 surface, with (2 Ang vacuum)BCA(1 Ang vacuum)ABC(1 Ang vacuum)
+    >>> fcc_slab(alat, atoms, "111", layers=' BCA ABC ', vacuum=(2, 1))
+
+    111 periodic structure with ABC(20 Ang vacuum)BC
+    The unit cell parameters will be periodic in this case, and it will not be
+    a slab.
+    >>> fcc_slab(alat, atoms, "111", layers='ABC BC', vacuum=20.)
```

Comment:
I think we should add one more example showing the orthogonal cell and use of `repeat`:
```
111 surface in an orthogonal (4x5) cell, maintaining the atom ordering
according to `lattice=[2, 1, 0]`:
>>> fcc_slab(alat, atoms, "111", orthogonal=True).repeat(5, axis=1).repeat(4, axis=0)
```